grin/src/adapters.rs

```diff
@@ -295,4 +296,12 @@ impl pool::BlockChain for PoolToChainAdapter {
 	fn get_unspent(&self, output_ref: &Commitment) -> Option<Output> {
 		self.chain.borrow().get_unspent(output_ref)
 	}
+
+	fn get_block_header_by_output_commit(&self, commit: &Commitment) -> Option<BlockHeader> {
```

Comment:
Same observation regarding the use of Option here. The pool::BlockChain trait should be modified to include some error type.